src/core/public/chrome/chrome_service.tsx

```diff
@@ -111,13 +108,12 @@ export class ChromeService {
   private initVisibility(application: StartDeps['application']) {
     // Start off the chrome service hidden if "embed" is in the hash query string.
     const isEmbedded = 'embed' in parse(location.hash.slice(1), true).query;
+    this.isForceHidden$ = new BehaviorSubject(isEmbedded);
 
-    this.toggleHidden$ = new BehaviorSubject(isEmbedded);
-    this.appHidden$ = merge(
-      // Default the app being hidden to the same value initial value as the chrome visibility
-      // in case the application service has not emitted an app ID yet, since we want to trigger
-      // combineLatest below regardless of having an application value yet.
-      of(isEmbedded),
+    const appHidden$ = merge(
+      // For the isVisible$ logic, having no mounted app is equivalent to having an hidden app
+      // in the sense that the chrome UI should now be displayed until an non-chromeless app is mounting or mounted
+      of(true),
```

Comment:
TIL, the chrome components actually need an initial `isVisible$` value, as they are present before the initial app is loaded/mounted. Removing this initial emission resulted in the same FOC issue, as these components are assuming the default value of `isVisible$` is true until a value is emitted. 

As we are now firing `application.currentAppId$` just before the app is mounted, it seems that this is an acceptable fix, as the chrome UI will be hidden until we start mounting the initial app (if it is not a chromeless app).

The rest of the logic remains unchanged.